specutils/io/default_loaders/jwst_reader.py

```diff
@@ -155,18 +233,24 @@ def _jwst_x1d_loader(file_obj, **kwargs):
             # SRCTYPE used to be in primary header, but was moved around. As
             # per most recent pipeline definition, it should be in the
             # EXTRACT1D extension.
+            #
+            # SRCTYPE should either be POINT or EXTENDED.  In some cases, it is UNKNOWN.
+            # if UNKNOWN, default to using the FLUX column.  Error out only when SRCTYPE
+            # cannot be found.
             srctype = None
             if "srctype" in hdu.header:
-                srctype = hdu.header.get("srctype")
+                srctype = hdu.header.get("srctype", None)
 
             if srctype == "POINT":
                 flux = Quantity(data["FLUX"])
                 uncertainty = StdDevUncertainty(data["ERROR"])
-
             elif srctype == "EXTENDED":
                 flux = Quantity(data["SURF_BRIGHT"])
                 uncertainty = StdDevUncertainty(hdu.data["SB_ERROR"])
-
+            elif srctype == 'UNKNOWN':
+                flux = Quantity(data["FLUX"])
+                uncertainty = StdDevUncertainty(data["ERROR"])
+                log.warning('SRCTYPE is UNKNOWN.  Defaulting to loading FLUX and ERROR data.')
             else:
                 raise RuntimeError(f"Keyword SRCTYPE is {srctype}.  It should "
                                    "be 'POINT' or 'EXTENDED'. Can't decide between `flux` and "
```

Comment:
There are cases where *no* `SRCTYPE` keyword exists at all in the header (there is a pemding Jira bug fix ticket). Shouldn't the `flux` variable be defined in the same way for the cases `SRCTYPE=UNKNOWN`, and `SRCTYPE=None`?

Reply:
that's a good question, which I did consider.  Talking to some pipeline folks, it seems that the `SRCTYPE` keyword should never be missing, but we've certainly found bugs in files where that was the case.  So I was originally thinking to leave the `raise` error to catch files where the `SRCTYPE` keyword is missing.   But I think it depends a bit on what we want Jdaviz or specutils to do when a user tries to load these files.  Should Jdaviz/specutils throw an error or still load them using some default logic?  I don't have a strong preference either way.   

If we switched to loading the file with some defaults, as we do with UNKNOWN, I could issue a new warning in the log.  In those cases, is it possible to catch these warnings and issue a warning snackbar message in Jdaviz ?

Reply:
I don't know if it is possible to catch warnings issued via the `logging` package. A possible workaround would be to have the code raise a custom exception, and have `jdaviz` code to catch that specific type of exception.

Reply:
In our tagup today, it was clarified that a decision was made a while ago (in the context of the NIRISS parser, PR#394) that the `jdaviz` tools should load the files anyway, even when no `SRCTYPE` keyword is present. In that case, a warning message of some sort (snackbar perhaps) could inform the user that the file is not compliant but is being loaded anyway using a default setting. The `MOSViz NIRISS parser uses a default of `EXTENDED`.

Reply:
Ok, sounds good.  I also clarified that idea on the tag up today.  I think we can default to loading something and issue a warning.  I'll update the code and ping when you it's ready.

Reply:
@ibusko @rosteen I've updated the code and pushed the changes.  The loader now sets the default `SRCTPYE`  for missing or UNKNOWN values to `POINT`.

Reply:
I wonder if this will generate user confusion with users, since the MOSViz NIRISS parser uses a default of `EXTENDED`. https://github.com/spacetelescope/jdaviz/blob/main/jdaviz/configs/mosviz/plugins/parsers.py

Reply:
Does quietly setting the `SRCTYPE` to "EXTENDED"  in Mosviz affect anything such that the user would even notice it was done? I don't think we need to worry about that here, since this explicitly warns the user about what it's doing (whereas in Mosviz the parser _knows_ what the sourcetype for NIRISS data should be and just silently adds it if needed without the user needing to worry about it). Seems like the warning that something might be wrong with the way the file was constructed is more important than the choice of what to use for the missing keyword.